### DISCUSSION TITLE
SOAP Services not exposed

### DISCUSSION THREAD

**Discussion by t1m0** [quarkiverse/quarkus-cxf#365] [Q&A] [answered]
Hi,

I tried to replicate the example with a small project, however I seem to miss something.
The SOAP-Service isn't getting exposed, however I see that the cxf extension is present.

I would have expected a similar log level to this:
`2022-02-21 07:20:03,710 INFO [io.qua.cxf.tra.CxfHandler] (Quarkus Main Thread) Web Service io.quarkiverse.it.cxf.GreetingWebServiceImpl on /soap available.`

https://github.com/t1m0/quarkus-soap-poc 

Maybe someone sees what I missed.

Thanks 
Timo

**Comment by shumonsharif** [comment, marked as answer]
Hi @t1m0 We've had some issues in the past relating to the API and implementation classes being in separate JARs.  

Taking a quick look at your repo, I noticed you aren't creating a Jandex index in your `pom.xml` files.  You may want to look into that, either using the maven plugin or you can also just define in your `application.properties`, I believe that may address your issue.  See below for further details on CDI bean discovery.

https://quarkus.io/guides/cdi-reference#bean_discovery

**Reply by t1m0** [reply]
Hi @shumonsharif  thanks for your response - we also just noticed this issue.
This was the problem.
